server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/ParquetResolver.java

```diff
@@ -356,4 +452,22 @@ private OneField resolveField(Group group, int columnIndex) {
         }
         return field;
     }
+
+    /**
+     * Sets configuration variables based on server configuration properties of pxf.parquet.write.decimal.overflow.
+     *
+     * @param configuration contains server configuration properties
+     */
+    public void parseDecimalOverflowOption(Configuration configuration) {
+        String decimalOverflowOption = configuration.get(PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_PROPERTY_NAME, PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ROUND).toLowerCase();
+
+        if (decimalOverflowOption.equals(PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ERROR)) {
+            isDecimalOverflowOptionError = true;
+        } else if (decimalOverflowOption.equals(PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ROUND)) {
+            isDecimalOverflowOptionRound = true;
+        } else if (!decimalOverflowOption.equals(PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_IGNORE)) {
+            throw new UnsupportedTypeException(String.format("Invalid configuration value %s for " +
+                    "pxf.parquet.write.decimal.overflow. Valid values are error, round, and ignore.", decimalOverflowOption));
+        }
```

Comment:
I'd use a switch here: https://docs.oracle.com/javase/8/docs/technotes/guides/language/strings-switch.html

```suggestion
        switch(configuration.get(PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_PROPERTY_NAME, PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ROUND).toLowerCase()) {
        case PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ERROR:
             isDecimalOverflowOptionError = true;
             break;
         case PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_ROUND:
              isDecimalOverflowOptionRound = true;
              break;
          case PXF_PARQUET_WRITE_DECIMAL_OVERFLOW_OPTION_IGNORE:
              break;
          default:
               throw new UnsupportedTypeException(String.format("Invalid configuration value %s for " +
                    "pxf.parquet.write.decimal.overflow. Valid values are error, round, and ignore.", decimalOverflowOption));
        }
```